core/src/main/java/c/org/rajawali3d/control/RenderControl.java

```diff
@@ -0,0 +1,212 @@
+     *
+     * @param runnable
+     */
+    void queueToMainThread(Runnable runnable);
+
+    /**
+     *
+     * @param runnable
+     */
+    void queueToRenderThread(Runnable runnable);
+
+    /**
+     *
+     * @param renderTask
+     */
+    void queueRenderTask(@NonNull RenderTask renderTask);
+
+    //
+    // Frame generation
+    //
+
+    /**
+     *  Flag value indicating use of the display refresh rate as the render frame rate
+     */
+    double USE_DISPLAY_REFRESH_RATE = Double.NaN;
+
+    /**
+     * Flag value indicating use of continuous rendering as fast as possible, rather than a periodic frame rate
+     */
+    double USE_CONTINUOUS_RENDERING = 0d;
+
+    /**
+     * Gets the screen refresh rate for the default display in frames per second.
+     *
+     * @return {@code double} The display refresh rate.
+     */
+    double getDisplayRefreshRate();
+
+    /**
+     * Sets the current target frameRate.
+     *
+     * @param frameRate {@code double} target frame frameRate; Double.NaN flags use of window refresh rate, 0 flags
+     *                                continuous rendering as fast as possible, >0 sets target frames per second
+     */
+    void setFrameRate(double frameRate);
+
+    /**
+     * Gets the current target frame rate in frames per second. Initial default is the display refresh rate.
+     *
+     * @return {@code double} The target frame rate.
+     */
+    double getFrameRate();
+
+    /**
+     * Checks whether frame processing is currently active (started but not yet stopped). This tracks the
+     * (paused/resumed) state of the underlying render thread.
+     *
+     * @return {@code true} if frame processing is active, {@code false} if frame processing is inactive
+     */
+    boolean areFramesActive();
+
+    /**
+     * Gets the system time of the most recent start of frame processing (render thread resume);
+     * 0 if frames are stopped (render thread paused)
+     *
+     * @return {@code long} System time in nanoseconds of most recent start of frame processing
+     */
+    long getFramesStartTime();
+
+    /**
+     * Gets the time elapsed since the most recent start of frame processing (render thread resume);
+     * 0 if frame are stopped (render thread paused)
+     *
+     * @return {code long} nanoseconds elapsed since most recent start of frame processing
+     */
+    long getFramesElapsedTime();
+
+    //
+    // Scenes population
+    //
+
+    /**
+     * Adds a {@link Scene} to the list of Scene frame delegates. The {@link Scene} will be notified via
+     * {@link Scene#initialize()}, and should take any needed actions.
+     *
+     * @param scene A {@link Scene} to add; duplicates are quietly ignored.
+     */
+    @RequiresRenderTask
+    void addScene(@NonNull Scene scene);
+
+    /**
+     * Removes a {@link Scene} from the list of Scene frame delegates. The {@link Scene} will be notified via
+     * {@link Scene#destroy()}, and should take any needed actions.
```

Comment:
I know I'm responsible for this but I think we should consider changing this contract to be `Scene#removed()` instead of implying that removal equals destruction.

Reply:
I'm not sure the callback is even necessary. I'm leaning towards adding/removing being internal operations that do the GL resource management under the covers, and letting the client do whatever they want before or after adding or removing.  In other words, `addToRenderControl()` means allocate/configure whatever GL resources are necessary internally to start using the Scene at the next frame draw, and `removeFromRenderControl()` means release whatever resources a Scene is currently using because it will not longer participate in frame draws.  That's why `onAddToRenderControl()` is part of the internal interface, the client shouldn't have to do anything

Reply:
Ok, let's see what this fleshes out to.

Reply:
Yeah, it's taken me a while to disentangle the old single-scene model lifecycle semantics from my brain, but I think its about there. 

Of course, I've said similar things before...